In [3]:
import pandas as pd
import re

This dictionary has the purpose to understand better the labels type of data. It will be based on all of the ``67 208 labels``.
- We will see if there is indeed only what would be considered as ``number, adresses and url``. 
- The frequency of those 
- The patterns of those 3 category (how many numbers per phone number label ; what are the different format of adresses ; what is an url looking like)

Findings : 

Part 1:
- Even though the participant_id go up to 254, there is only 94 individual participant Signer. 
- 31.7% of the labels are composed of a number followed with a space without being a phone number (21 344 out of 67 208)
- 24.3% of the labels are composed of ``only numbers`` (including + - )  (16 356 out of 67 208)
- 16.2% of the labels are composed of www, http, .com, .net (10 886 out of 67 208)
- 12.4% of the labels are composed of ``only letters``. (8 347 out of 67 208)

84.6% of the labels classified. 

only 10 275 out of 67 208 are not classified after those criteria. to look at them it is : df4

Part 2: 
- ``The dictionary of frequency for each character is done on the bottom of the page``

<!-- This is at the top of the file -->
[Go to Bottom](#bottom)

In [3]:
df = pd.read_csv('d:/Bureau/ASLCapstone/train.csv/train.csv')
df

path     file_id  sequence_id  \
0         train_landmarks/5414471.parquet     5414471   1816796431   
1         train_landmarks/5414471.parquet     5414471   1816825349   
2         train_landmarks/5414471.parquet     5414471   1816909464   
3         train_landmarks/5414471.parquet     5414471   1816967051   
4         train_landmarks/5414471.parquet     5414471   1817123330   
...                                   ...         ...          ...   
67203  train_landmarks/2118949241.parquet  2118949241    388192924   
67204  train_landmarks/2118949241.parquet  2118949241    388225542   
67205  train_landmarks/2118949241.parquet  2118949241    388232076   
67206  train_landmarks/2118949241.parquet  2118949241    388235284   
67207  train_landmarks/2118949241.parquet  2118949241    388332538   

       participant_id                          phrase  
0                 217                    3 creekhouse  
1                 107                 scales/kuhaylah  
2                   1             1383 william lanier  
3                  63               988 franklin lane  
4                  89       6920 northeast 661st road  
...               ...                             ...  
67203              88                    431-366-2913  
67204             154                    994-392-3850  
67205              95  https://www.tianjiagenomes.com  
67206              36               90 kerwood circle  
67207             176                      802 co 66b  

[67208 rows x 5 columns]

In [4]:
# Assuming your dataset is stored in a pandas DataFrame named 'df'
# Replace 'participant_id' with the actual column name containing participant IDs
unique_participant_count = df['participant_id'].nunique()

print("Number of different participant_id:", unique_participant_count)


Number of different participant_id: 94


In [5]:
distinct_participant_ids = df['participant_id'].unique()
distinct_participant_ids.sort()

print(distinct_participant_ids)

[  0   1   2   4   6   9  10  13  15  18  20  21  24  25  27  33  36  38
  40  43  53  56  59  63  68  70  71  72  73  74  76  80  81  88  89  92
  93  95 102 105 107 109 112 113 115 117 121 122 125 128 135 136 138 141
 143 145 147 151 153 154 157 158 159 160 161 168 169 171 176 178 181 186
 187 188 192 196 202 203 216 217 219 223 225 227 230 231 233 236 239 241
 242 246 251 254]


## First step: Let's see the biggest chunk of data we would be easily able to group together. 

In [6]:
# We'll use regular expressions to filter rows with only numbers and no letters
filtered_rows0 = df[df['phrase'].str.contains(r'^\d+$')]
filtered_rows0

path     file_id  sequence_id  \
184       train_landmarks/5414471.parquet     5414471   1823097602   
509       train_landmarks/5414471.parquet     5414471   1832431491   
762       train_landmarks/5414471.parquet     5414471   1840377388   
819       train_landmarks/5414471.parquet     5414471   1842341816   
959       train_landmarks/5414471.parquet     5414471   1847445336   
...                                   ...         ...          ...   
64695  train_landmarks/2072296290.parquet  2072296290     81596268   
65004  train_landmarks/2072296290.parquet  2072296290     91772699   
65285  train_landmarks/2072876091.parquet  2072876091   1947709366   
65770  train_landmarks/2072876091.parquet  2072876091   1961810650   
66566  train_landmarks/2118949241.parquet  2118949241    366855951   

       participant_id   phrase  
184               113   322019  
509                88  2085199  
762               105   449517  
819               178   213476  
959               125    19546  
...               ...      ...  
64695              93  2428445  
65004             239  1742392  
65285             105   149887  
65770              18   322019  
66566              89    17872  

[103 rows x 5 columns]

In [9]:
# To filter rows with phrases containing no letters:
filtered_rows = df[df['phrase'].str.contains(r'^[^A-Za-z]+$')]

# The 'filtered_rows' DataFrame will now contain only the rows where the 'phrase' column contains no letters.

# To see the filtered DataFrame:
filtered_rows.tail(60)

path     file_id  sequence_id  \
66936  train_landmarks/2118949241.parquet  2118949241    379343212   
66937  train_landmarks/2118949241.parquet  2118949241    379372506   
66938  train_landmarks/2118949241.parquet  2118949241    379448331   
66939  train_landmarks/2118949241.parquet  2118949241    379486571   
66941  train_landmarks/2118949241.parquet  2118949241    379511535   
66952  train_landmarks/2118949241.parquet  2118949241    380057478   
66959  train_landmarks/2118949241.parquet  2118949241    380243251   
66960  train_landmarks/2118949241.parquet  2118949241    380262140   
66967  train_landmarks/2118949241.parquet  2118949241    380492852   
66971  train_landmarks/2118949241.parquet  2118949241    380590181   
66976  train_landmarks/2118949241.parquet  2118949241    380713808   
66982  train_landmarks/2118949241.parquet  2118949241    380919982   
66984  train_landmarks/2118949241.parquet  2118949241    380976459   
66987  train_landmarks/2118949241.parquet  2118949241    381131617   
66990  train_landmarks/2118949241.parquet  2118949241    381276523   
66995  train_landmarks/2118949241.parquet  2118949241    381408660   
67001  train_landmarks/2118949241.parquet  2118949241    381649766   
67004  train_landmarks/2118949241.parquet  2118949241    381668603   
67009  train_landmarks/2118949241.parquet  2118949241    381922106   
67010  train_landmarks/2118949241.parquet  2118949241    381942677   
67011  train_landmarks/2118949241.parquet  2118949241    381945622   
67017  train_landmarks/2118949241.parquet  2118949241    382120200   
67021  train_landmarks/2118949241.parquet  2118949241    382158177   
67023  train_landmarks/2118949241.parquet  2118949241    382208469   
67024  train_landmarks/2118949241.parquet  2118949241    382332451   
67026  train_landmarks/2118949241.parquet  2118949241    382388939   
67028  train_landmarks/2118949241.parquet  2118949241    382544044   
67029  train_landmarks/2118949241.parquet  2118949241    382747608   
67040  train_landmarks/2118949241.parquet  2118949241    383080386   
67056  train_landmarks/2118949241.parquet  2118949241    383523015   
67058  train_landmarks/2118949241.parquet  2118949241    383549909   
67068  train_landmarks/2118949241.parquet  2118949241    383759468   
67080  train_landmarks/2118949241.parquet  2118949241    383933655   
67084  train_landmarks/2118949241.parquet  2118949241    383995388   
67096  train_landmarks/2118949241.parquet  2118949241    384468367   
67102  train_landmarks/2118949241.parquet  2118949241    384593910   
67115  train_landmarks/2118949241.parquet  2118949241    384953176   
67117  train_landmarks/2118949241.parquet  2118949241    385019033   
67118  train_landmarks/2118949241.parquet  2118949241    385066882   
67130  train_landmarks/2118949241.parquet  2118949241    385206000   
67132  train_landmarks/2118949241.parquet  2118949241    385214120   
67139  train_landmarks/2118949241.parquet  2118949241    385614157   
67141  train_landmarks/2118949241.parquet  2118949241    385772070   
67142  train_landmarks/2118949241.parquet  2118949241    385807843   
67149  train_landmarks/2118949241.parquet  2118949241    386200031   
67150  train_landmarks/2118949241.parquet  2118949241    386209059   
67153  train_landmarks/2118949241.parquet  2118949241    386361726   
67158  train_landmarks/2118949241.parquet  2118949241    386603022   
67162  train_landmarks/2118949241.parquet  2118949241    386736891   
67163  train_landmarks/2118949241.parquet  2118949241    386822961   
67167  train_landmarks/2118949241.parquet  2118949241    386889882   
67171  train_landmarks/2118949241.parquet  2118949241    387033750   
67177  train_landmarks/2118949241.parquet  2118949241    387257578   
67189  train_landmarks/2118949241.parquet  2118949241    387477535   
67190  train_landmarks/2118949241.parquet  2118949241    387496458   
67197  train_landmarks/2118949241.parquet  2118949241    387926412   
67198  train_landmarks/2118949241.parquet  2118

In [15]:
# We'll use regular expressions to filter rows with only letters (or space)
filtered_rows1 = df[df['phrase'].str.contains(r'^[a-zA-Z\s]+$')]
filtered_rows1

path     file_id  sequence_id  \
10        train_landmarks/5414471.parquet     5414471   1817221592   
16        train_landmarks/5414471.parquet     5414471   1817362238   
19        train_landmarks/5414471.parquet     5414471   1817438012   
27        train_landmarks/5414471.parquet     5414471   1817720859   
30        train_landmarks/5414471.parquet     5414471   1817781680   
...                                   ...         ...          ...   
67173  train_landmarks/2118949241.parquet  2118949241    387070237   
67183  train_landmarks/2118949241.parquet  2118949241    387315309   
67188  train_landmarks/2118949241.parquet  2118949241    387474772   
67191  train_landmarks/2118949241.parquet  2118949241    387524930   
67195  train_landmarks/2118949241.parquet  2118949241    387888464   

       participant_id              phrase  
10                 27    federico pearson  
16                231         leona owens  
19                202         kati castro  
27                102            tampa fl  
30                254      jami mcfarland  
...               ...                 ...  
67173             217  angelica mcfarland  
67183             254           javier le  
67188              95      nathanial hale  
67191             151        darcy hanson  
67195               1              jan li  

[8347 rows x 5 columns]

## To have an overview of the rest of the data, let's remove those from the data into another data frame we will call df1. This way we will look to find some patterns to explore.

In [19]:
# To create a new DataFrame that removes cases where 'phrase' contains only letters:
df_no_letters = df[~df['phrase'].str.contains(r'^[a-zA-Z\s]+$')]

# To create a new DataFrame that removes cases where 'phrase' contains no letters:
df_with_letters = df[~df['phrase'].str.contains(r'^[^A-Za-z]+$')]

# To combine both conditions and create the final DataFrame that removes both cases:
df1 = df_no_letters[df_no_letters['phrase'].isin(df_with_letters['phrase'])]

# To display the final DataFrame:
df1.head(50)

path  file_id  sequence_id  participant_id  \
0   train_landmarks/5414471.parquet  5414471   1816796431             217   
1   train_landmarks/5414471.parquet  5414471   1816825349             107   
2   train_landmarks/5414471.parquet  5414471   1816909464               1   
3   train_landmarks/5414471.parquet  5414471   1816967051              63   
4   train_landmarks/5414471.parquet  5414471   1817123330              89   
5   train_landmarks/5414471.parquet  5414471   1817141095              38   
6   train_landmarks/5414471.parquet  5414471   1817169529              70   
7   train_landmarks/5414471.parquet  5414471   1817171518             202   
9   train_landmarks/5414471.parquet  5414471   1817216847              93   
11  train_landmarks/5414471.parquet  5414471   1817221621             153   
12  train_landmarks/5414471.parquet  5414471   1817229827             236   
15  train_landmarks/5414471.parquet  5414471   1817348598              20   
17  train_landmarks/5414471.parquet  5414471   1817370426              21   
20  train_landmarks/5414471.parquet  5414471   1817581380              15   
21  train_landmarks/5414471.parquet  5414471   1817606833             254   
22  train_landmarks/5414471.parquet  5414471   1817631401             169   
24  train_landmarks/5414471.parquet  5414471   1817657937               2   
26  train_landmarks/5414471.parquet  5414471   1817711982              70   
28  train_landmarks/5414471.parquet  5414471   1817740422             141   
29  train_landmarks/5414471.parquet  5414471   1817770333             109   
31  train_landmarks/5414471.parquet  5414471   1817808989             158   
32  train_landmarks/5414471.parquet  5414471   1817815567             230   
33  train_landmarks/5414471.parquet  5414471   1817882602             254   
35  train_landmarks/5414471.parquet  5414471   1817940885               0   
36  train_landmarks/5414471.parquet  5414471   1817947249             109   
37  train_landmarks/5414471.parquet  5414471   1817957149             105   
38  train_landmarks/5414471.parquet  5414471   1817969676             236   
39  train_landmarks/5414471.parquet  5414471   1818029273             217   
41  train_landmarks/5414471.parquet  5414471   1818053931             230   
42  train_landmarks/5414471.parquet  5414471   1818125771             105   
43  train_landmarks/5414471.parquet  5414471   1818154356             186   
44  train_landmarks/5414471.parquet  5414471   1818160881              89   
45  train_landmarks/5414471.parquet  5414471   1818211300             202   
46  train_landmarks/5414471.parquet  5414471   1818216000              56   
50  train_landmarks/5414471.parquet  5414471   1818249176             231   
55  train_landmarks/5414471.parquet  5414471   1818613605             136   
56  train_landmarks/5414471.parquet  5414471   1818674500             196   
58  train_landmarks/5414471.parquet  5414471   1818700520             236   
59  train_landmarks/5414471.parquet  5414471   1818712825             225   
60  train_landmarks/5414471.parquet  5414471   1818726340             186   
61  train_landmarks/5414471.parquet  5414471   1818745682              24   
62  train_landmarks/5414471.parquet  5414471   1818787239             161   
63  train_landmarks/5414471.parquet  5414471   1818802303             230   
64  train_landmarks/5414471.parquet  5414471   1818807591             242   
65  train_landmarks/5414471.parquet  5414471   1818863391               0   
66  train_landmarks/5414471.parquet  5414471   1818864430               1   
67  train_landmarks/5414471.parquet  5414471   1818872649             136   
69  train_landmarks/5414471.parquet  5414471   1818887623             168   
70  train_landmarks/5414471.parquet  5414471   1818895313               9   
72  train_landmarks/5414471.parquet  5414471   1818992993             216   

                            phrase  
0                     3 creekhouse  
1                  scales/kuh

#### Safety check : 67 208 -8347 - 16356 = 42 505 ; Good.

In [24]:
# To display every row where 'phrase' contains 'www':
rows_with_www1 = df1[df1['phrase'].str.contains('www', case=False)]

# To display the resulting rows:
print(rows_with_www1)

                                     path     file_id  sequence_id  \
5         train_landmarks/5414471.parquet     5414471   1817141095   
31        train_landmarks/5414471.parquet     5414471   1817808989   
37        train_landmarks/5414471.parquet     5414471   1817957149   
42        train_landmarks/5414471.parquet     5414471   1818125771   
61        train_landmarks/5414471.parquet     5414471   1818745682   
...                                   ...         ...          ...   
67159  train_landmarks/2118949241.parquet  2118949241    386605418   
67165  train_landmarks/2118949241.parquet  2118949241    386858900   
67174  train_landmarks/2118949241.parquet  2118949241    387085237   
67202  train_landmarks/2118949241.parquet  2118949241    388105954   
67205  train_landmarks/2118949241.parquet  2118949241    388232076   

       participant_id                          phrase  
5                  38                 www.freem.ne.jp  
31                158       www.horseillustrate

Let's make a supplementary safety check and see if it is indeed similar to the initial df.

In [23]:
# To display every row where 'phrase' contains 'www':
rows_with_www = df[df['phrase'].str.contains('www', case=False)]

# To display the resulting rows:
print(rows_with_www)

                                     path     file_id  sequence_id  \
5         train_landmarks/5414471.parquet     5414471   1817141095   
31        train_landmarks/5414471.parquet     5414471   1817808989   
37        train_landmarks/5414471.parquet     5414471   1817957149   
42        train_landmarks/5414471.parquet     5414471   1818125771   
61        train_landmarks/5414471.parquet     5414471   1818745682   
...                                   ...         ...          ...   
67159  train_landmarks/2118949241.parquet  2118949241    386605418   
67165  train_landmarks/2118949241.parquet  2118949241    386858900   
67174  train_landmarks/2118949241.parquet  2118949241    387085237   
67202  train_landmarks/2118949241.parquet  2118949241    388105954   
67205  train_landmarks/2118949241.parquet  2118949241    388232076   

       participant_id                          phrase  
5                  38                 www.freem.ne.jp  
31                158       www.horseillustrate

In [26]:
# To display every row where 'phrase' contains 'www' or 'http':
rows_with_www_http = df1[df1['phrase'].str.contains('www|http', case=False)]

# To count the occurrences of 'www' or 'http':
count_www_http = len(rows_with_www_http)

# To display the resulting rows:
print("Rows containing 'www' or 'http':")
print(rows_with_www_http)

# To display the count:
print("Count of occurrences of 'www' or 'http':", count_www_http)

Rows containing 'www' or 'http':
                                     path     file_id  sequence_id  \
5         train_landmarks/5414471.parquet     5414471   1817141095   
6         train_landmarks/5414471.parquet     5414471   1817169529   
31        train_landmarks/5414471.parquet     5414471   1817808989   
37        train_landmarks/5414471.parquet     5414471   1817957149   
42        train_landmarks/5414471.parquet     5414471   1818125771   
...                                   ...         ...          ...   
67170  train_landmarks/2118949241.parquet  2118949241    387021660   
67174  train_landmarks/2118949241.parquet  2118949241    387085237   
67199  train_landmarks/2118949241.parquet  2118949241    387993269   
67202  train_landmarks/2118949241.parquet  2118949241    388105954   
67205  train_landmarks/2118949241.parquet  2118949241    388232076   

       participant_id                          phrase  
5                  38                 www.freem.ne.jp  
6             

The majority of the website have both www and http in our labels. (6118 with www only, and 7093 with www and http)

In [32]:
# To remove rows containing 'www' or 'http' in any column:
df2 = df1[~df1.apply(lambda x: x.astype(str).str.contains('www|http')).any(axis=1)]

# To display the resulting DataFrame after filtering:
df2.tail(50)

path     file_id  sequence_id  \
67111  train_landmarks/2118949241.parquet  2118949241    384915497   
67112  train_landmarks/2118949241.parquet  2118949241    384933586   
67113  train_landmarks/2118949241.parquet  2118949241    384936621   
67114  train_landmarks/2118949241.parquet  2118949241    384939486   
67120  train_landmarks/2118949241.parquet  2118949241    385068751   
67121  train_landmarks/2118949241.parquet  2118949241    385076838   
67122  train_landmarks/2118949241.parquet  2118949241    385092589   
67123  train_landmarks/2118949241.parquet  2118949241    385095495   
67125  train_landmarks/2118949241.parquet  2118949241    385107535   
67126  train_landmarks/2118949241.parquet  2118949241    385108408   
67128  train_landmarks/2118949241.parquet  2118949241    385183905   
67129  train_landmarks/2118949241.parquet  2118949241    385195459   
67133  train_landmarks/2118949241.parquet  2118949241    385243069   
67134  train_landmarks/2118949241.parquet  2118949241    385290906   
67135  train_landmarks/2118949241.parquet  2118949241    385436924   
67136  train_landmarks/2118949241.parquet  2118949241    385454191   
67137  train_landmarks/2118949241.parquet  2118949241    385459660   
67143  train_landmarks/2118949241.parquet  2118949241    385818990   
67144  train_landmarks/2118949241.parquet  2118949241    385867160   
67145  train_landmarks/2118949241.parquet  2118949241    385870153   
67146  train_landmarks/2118949241.parquet  2118949241    385912830   
67147  train_landmarks/2118949241.parquet  2118949241    385919340   
67148  train_landmarks/2118949241.parquet  2118949241    386151815   
67152  train_landmarks/2118949241.parquet  2118949241    386331184   
67154  train_landmarks/2118949241.parquet  2118949241    386466332   
67156  train_landmarks/2118949241.parquet  2118949241    386515061   
67157  train_landmarks/2118949241.parquet  2118949241    386546915   
67160  train_landmarks/2118949241.parquet  2118949241    386675746   
67161  train_landmarks/2118949241.parquet  2118949241    386703499   
67168  train_landmarks/2118949241.parquet  2118949241    386908676   
67169  train_landmarks/2118949241.parquet  2118949241    387011414   
67172  train_landmarks/2118949241.parquet  2118949241    387033856   
67175  train_landmarks/2118949241.parquet  2118949241    387128152   
67176  train_landmarks/2118949241.parquet  2118949241    387214048   
67178  train_landmarks/2118949241.parquet  2118949241    387274713   
67179  train_landmarks/2118949241.parquet  2118949241    387289656   
67180  train_landmarks/2118949241.parquet  2118949241    387309753   
67181  train_landmarks/2118949241.parquet  2118949241    387312071   
67182  train_landmarks/2118949241.parquet  2118949241    387315228   
67184  train_landmarks/2118949241.parquet  2118949241    387336292   
67185  train_landmarks/2118949241.parquet  2118949241    387382391   
67186  train_landmarks/2118949241.parquet  2118949241    387389076   
67187  train_landmarks/2118949241.parquet  2118949241    387390786   
67192  train_landmarks/2118949241.parquet  2118949241    387639942   
67193  train_landmarks/2118949241.parquet  2118949241    387825647   
67194  train_landmarks/2118949241.parquet  2118949241    387839640   
67196  train_landmarks/2118949241.parquet  2118949241    387891436   
67200  train_landmarks/2118949241.parquet  2118949241    387998966   
67206  train_landmarks/2118949241.parquet  2118949241    388235284   
67207  train_landmarks/2118949241.parquet  2118949241    388332538   

       participant_id                          phrase  
67111              76        dandong_zlgl/the_big_lie  
67112             145          imix-beauty.cz/pevitsy  
67113              81       8076 street tammany court  
67114             176       bardot-brittany-jumpsuit/  
67120             109                5860 geggen tina  
67121              72              9989 serenity hill  
67122               1          cse.eng.marmara.edu.tr  
67123             

#### Safety check, 42 505 - 7093 = 35 412 ; Good.

In [35]:
# To display every row where 'phrase' contains '.net', '.com', or '.fr':
rows_with_keywords = df2[df2['phrase'].str.contains('.net|.com', case=False)]

# To display the resulting rows:
print(rows_with_keywords)

                                     path     file_id  sequence_id  \
43        train_landmarks/5414471.parquet     5414471   1818154356   
79        train_landmarks/5414471.parquet     5414471   1819309566   
99        train_landmarks/5414471.parquet     5414471   1819753051   
115       train_landmarks/5414471.parquet     5414471   1820295143   
144       train_landmarks/5414471.parquet     5414471   1821174920   
...                                   ...         ...          ...   
67086  train_landmarks/2118949241.parquet  2118949241    384146029   
67137  train_landmarks/2118949241.parquet  2118949241    385459660   
67152  train_landmarks/2118949241.parquet  2118949241    386331184   
67168  train_landmarks/2118949241.parquet  2118949241    386908676   
67194  train_landmarks/2118949241.parquet  2118949241    387839640   

       participant_id                         phrase  
43                186      shop.oliphan.com/tana-lea  
79                 20           tempemarketplace.

In [38]:
df3 = df2[~df2.apply(lambda x: x.astype(str).str.contains('.com|.net')).any(axis=1)]

# To display the resulting DataFrame after filtering:
df3.head(50)

path  file_id  sequence_id  participant_id  \
0   train_landmarks/5414471.parquet  5414471   1816796431             217   
1   train_landmarks/5414471.parquet  5414471   1816825349             107   
2   train_landmarks/5414471.parquet  5414471   1816909464               1   
3   train_landmarks/5414471.parquet  5414471   1816967051              63   
4   train_landmarks/5414471.parquet  5414471   1817123330              89   
7   train_landmarks/5414471.parquet  5414471   1817171518             202   
9   train_landmarks/5414471.parquet  5414471   1817216847              93   
11  train_landmarks/5414471.parquet  5414471   1817221621             153   
12  train_landmarks/5414471.parquet  5414471   1817229827             236   
15  train_landmarks/5414471.parquet  5414471   1817348598              20   
17  train_landmarks/5414471.parquet  5414471   1817370426              21   
20  train_landmarks/5414471.parquet  5414471   1817581380              15   
21  train_landmarks/5414471.parquet  5414471   1817606833             254   
22  train_landmarks/5414471.parquet  5414471   1817631401             169   
24  train_landmarks/5414471.parquet  5414471   1817657937               2   
26  train_landmarks/5414471.parquet  5414471   1817711982              70   
28  train_landmarks/5414471.parquet  5414471   1817740422             141   
29  train_landmarks/5414471.parquet  5414471   1817770333             109   
32  train_landmarks/5414471.parquet  5414471   1817815567             230   
33  train_landmarks/5414471.parquet  5414471   1817882602             254   
35  train_landmarks/5414471.parquet  5414471   1817940885               0   
36  train_landmarks/5414471.parquet  5414471   1817947249             109   
38  train_landmarks/5414471.parquet  5414471   1817969676             236   
39  train_landmarks/5414471.parquet  5414471   1818029273             217   
41  train_landmarks/5414471.parquet  5414471   1818053931             230   
44  train_landmarks/5414471.parquet  5414471   1818160881              89   
45  train_landmarks/5414471.parquet  5414471   1818211300             202   
46  train_landmarks/5414471.parquet  5414471   1818216000              56   
50  train_landmarks/5414471.parquet  5414471   1818249176             231   
55  train_landmarks/5414471.parquet  5414471   1818613605             136   
56  train_landmarks/5414471.parquet  5414471   1818674500             196   
58  train_landmarks/5414471.parquet  5414471   1818700520             236   
59  train_landmarks/5414471.parquet  5414471   1818712825             225   
60  train_landmarks/5414471.parquet  5414471   1818726340             186   
62  train_landmarks/5414471.parquet  5414471   1818787239             161   
63  train_landmarks/5414471.parquet  5414471   1818802303             230   
65  train_landmarks/5414471.parquet  5414471   1818863391               0   
66  train_landmarks/5414471.parquet  5414471   1818864430               1   
67  train_landmarks/5414471.parquet  5414471   1818872649             136   
69  train_landmarks/5414471.parquet  5414471   1818887623             168   
70  train_landmarks/5414471.parquet  5414471   1818895313               9   
72  train_landmarks/5414471.parquet  5414471   1818992993             216   
77  train_landmarks/5414471.parquet  5414471   1819185262             151   
78  train_landmarks/5414471.parquet  5414471   1819251883             157   
82  train_landmarks/5414471.parquet  5414471   1819405994               2   
83  train_landmarks/5414471.parquet  5414471   1819437603             121   
84  train_landmarks/5414471.parquet  5414471   1819442021              92   
85  train_landmarks/5414471.parquet  5414471   1819442901             225   
86  train_landmarks/5414471.parquet  5414471   1819492087              59   
87  train_landmarks/5414471.parquet  5414471   1819496260             115   

                            phrase  
0                     3 creekhouse  
1                  scales/kuh

In [41]:
# To display every row where 'phrase' contains 'web':
rows_with_web = df3[df3['phrase'].str.contains('web', case=False)]

# To display the resulting rows:
rows_with_web

path     file_id  sequence_id  \
86        train_landmarks/5414471.parquet     5414471   1819492087   
1019    train_landmarks/105143404.parquet   105143404   1785271605   
1894    train_landmarks/105143404.parquet   105143404   1813296777   
2191    train_landmarks/128822441.parquet   128822441    203463059   
4286    train_landmarks/152029243.parquet   152029243   1093711807   
5807    train_landmarks/169560558.parquet   169560558   1330458016   
5840    train_landmarks/169560558.parquet   169560558   1331598375   
6034    train_landmarks/175396851.parquet   175396851    100657742   
10722   train_landmarks/388576474.parquet   388576474   1076969364   
11036   train_landmarks/425182931.parquet   425182931   1118221084   
13257   train_landmarks/450474571.parquet   450474571   2147040558   
19265   train_landmarks/546816846.parquet   546816846   1463633744   
20310   train_landmarks/568753759.parquet   568753759    959350056   
21096   train_landmarks/568753759.parquet   568753759    983660655   
21174   train_landmarks/568753759.parquet   568753759    986237704   
22226   train_landmarks/614661748.parquet   614661748    541714278   
25561   train_landmarks/683666742.parquet   683666742   1695568154   
29516   train_landmarks/939623093.parquet   939623093   1760073574   
29954   train_landmarks/939623093.parquet   939623093   1773725445   
31681  train_landmarks/1021040628.parquet  1021040628   1733139557   
32223  train_landmarks/1021040628.parquet  1021040628   1750829876   
32841  train_landmarks/1098899348.parquet  1098899348   1867628077   
33923  train_landmarks/1099408314.parquet  1099408314   1230563432   
35277  train_landmarks/1134756332.parquet  1134756332    133140167   
36608  train_landmarks/1255240050.parquet  1255240050    584067201   
38627  train_landmarks/1341528257.parquet  1341528257   1410675918   
39694  train_landmarks/1358493307.parquet  1358493307    273110306   
40461  train_landmarks/1365275733.parquet  1365275733   2080370309   
40987  train_landmarks/1365275733.parquet  1365275733   2097627894   
43192  train_landmarks/1405046009.parquet  1405046009    258377359   
44772  train_landmarks/1497621680.parquet  1497621680   1038772759   
48095  train_landmarks/1562234637.parquet  1562234637    954059779   
48275  train_landmarks/1643479812.parquet  1643479812    389703025   
50380  train_landmarks/1662742697.parquet  1662742697   1342224294   
51065  train_landmarks/1662742697.parquet  1662742697   1363620755   
51510  train_landmarks/1664666588.parquet  1664666588    489314219   
52040  train_landmarks/1664666588.parquet  1664666588    505398007   
52339  train_landmarks/1726141437.parquet  1726141437   1244280647   
53267  train_landmarks/1785039512.parquet  1785039512   1623885532   
54475  train_landmarks/1865557033.parquet  1865557033   2113756727   
54673  train_landmarks/1865557033.parquet  1865557033   2119889951   
54811  train_landmarks/1865557033.parquet  1865557033   2124317804   
55073  train_landmarks/1865557033.parquet  1865557033   2133342286   
55259  train_landmarks/1880177496.parquet  1880177496    859673286   
55375  train_landmarks/1880177496.parquet  1880177496    863677360   
55794  train_landmarks/1880177496.parquet  1880177496    877707025   
59418  train_landmarks/1967755728.parquet  1967755728    995671520   
59861  train_landmarks/1967755728.parquet  1967755728   1009404475   
61519  train_landmarks/1997878546.parquet  1997878546    709304894   
61945  train_landmarks/1997878546.parquet  1997878546    722675435   
63423  train_landmarks/2036580525.parquet  2036580525   1374803688   
64109  train_landmarks/2036580525.parquet  2036580525   1396391211   
65215  train_landmarks/2072876091.parquet  2072876091   1944911383   

       participant_id                          phrase  
86                 59      webthesis.biblio.polito.it  
1019               24      /anthonygreen/servicos-web  
1894               27      /anthonygreen/servicos-web  
2191              105         /w

### We won't be able to define the labels having 'web' as url, as there are many adresses where 'web' is present. ps: let's see back later about it as still seemingly about 85% of the case it still work we might still count them as url later. 

In [44]:
df3.head(60)

path  file_id  sequence_id  participant_id  \
0    train_landmarks/5414471.parquet  5414471   1816796431             217   
1    train_landmarks/5414471.parquet  5414471   1816825349             107   
2    train_landmarks/5414471.parquet  5414471   1816909464               1   
3    train_landmarks/5414471.parquet  5414471   1816967051              63   
4    train_landmarks/5414471.parquet  5414471   1817123330              89   
7    train_landmarks/5414471.parquet  5414471   1817171518             202   
9    train_landmarks/5414471.parquet  5414471   1817216847              93   
11   train_landmarks/5414471.parquet  5414471   1817221621             153   
12   train_landmarks/5414471.parquet  5414471   1817229827             236   
15   train_landmarks/5414471.parquet  5414471   1817348598              20   
17   train_landmarks/5414471.parquet  5414471   1817370426              21   
20   train_landmarks/5414471.parquet  5414471   1817581380              15   
21   train_landmarks/5414471.parquet  5414471   1817606833             254   
22   train_landmarks/5414471.parquet  5414471   1817631401             169   
24   train_landmarks/5414471.parquet  5414471   1817657937               2   
26   train_landmarks/5414471.parquet  5414471   1817711982              70   
28   train_landmarks/5414471.parquet  5414471   1817740422             141   
29   train_landmarks/5414471.parquet  5414471   1817770333             109   
32   train_landmarks/5414471.parquet  5414471   1817815567             230   
33   train_landmarks/5414471.parquet  5414471   1817882602             254   
35   train_landmarks/5414471.parquet  5414471   1817940885               0   
36   train_landmarks/5414471.parquet  5414471   1817947249             109   
38   train_landmarks/5414471.parquet  5414471   1817969676             236   
39   train_landmarks/5414471.parquet  5414471   1818029273             217   
41   train_landmarks/5414471.parquet  5414471   1818053931             230   
44   train_landmarks/5414471.parquet  5414471   1818160881              89   
45   train_landmarks/5414471.parquet  5414471   1818211300             202   
46   train_landmarks/5414471.parquet  5414471   1818216000              56   
50   train_landmarks/5414471.parquet  5414471   1818249176             231   
55   train_landmarks/5414471.parquet  5414471   1818613605             136   
56   train_landmarks/5414471.parquet  5414471   1818674500             196   
58   train_landmarks/5414471.parquet  5414471   1818700520             236   
59   train_landmarks/5414471.parquet  5414471   1818712825             225   
60   train_landmarks/5414471.parquet  5414471   1818726340             186   
62   train_landmarks/5414471.parquet  5414471   1818787239             161   
63   train_landmarks/5414471.parquet  5414471   1818802303             230   
65   train_landmarks/5414471.parquet  5414471   1818863391               0   
66   train_landmarks/5414471.parquet  5414471   1818864430               1   
67   train_landmarks/5414471.parquet  5414471   1818872649             136   
69   train_landmarks/5414471.parquet  5414471   1818887623             168   
70   train_landmarks/5414471.parquet  5414471   1818895313               9   
72   train_landmarks/5414471.parquet  5414471   1818992993             216   
77   train_landmarks/5414471.parquet  5414471   1819185262             151   
78   train_landmarks/5414471.parquet  5414471   1819251883             157   
82   train_landmarks/5414471.parquet  5414471   1819405994               2   
83   train_landmarks/5414471.parquet  5414471   1819437603             121   
84   train_landmarks/5414471.parquet  5414471   1819442021              92   
85   train_landmarks/5414471.parquet  5414471   1819442901             225   
86   train_landmarks/5414471.parquet  5414471   1819492087              59   
87   train_landmarks/5414471.parquet  5414471   1819496260             115   
89   train_landmarks/5414471.parquet  5414471   181953

### From the head and tail 120 data it really seem like there is a pattern for what would be considered as an adress. 
``They start with a number followed by a space``. Let's see on a few hundred more rows to see if it's fiable. 

In [53]:
rows = df3.iloc[250:300]
rows

path  file_id  sequence_id  participant_id  \
529  train_landmarks/5414471.parquet  5414471   1832812547             216   
530  train_landmarks/5414471.parquet  5414471   1832813965             186   
531  train_landmarks/5414471.parquet  5414471   1832831944             178   
535  train_landmarks/5414471.parquet  5414471   1832933170             231   
536  train_landmarks/5414471.parquet  5414471   1832954665             128   
538  train_landmarks/5414471.parquet  5414471   1832995543             202   
541  train_landmarks/5414471.parquet  5414471   1833029749             196   
544  train_landmarks/5414471.parquet  5414471   1833166228             161   
545  train_landmarks/5414471.parquet  5414471   1833167208             246   
548  train_landmarks/5414471.parquet  5414471   1833220685              38   
549  train_landmarks/5414471.parquet  5414471   1833250342              81   
551  train_landmarks/5414471.parquet  5414471   1833298164             178   
556  train_landmarks/5414471.parquet  5414471   1833508348              88   
557  train_landmarks/5414471.parquet  5414471   1833564379             154   
560  train_landmarks/5414471.parquet  5414471   1833604808              89   
562  train_landmarks/5414471.parquet  5414471   1833667416              89   
564  train_landmarks/5414471.parquet  5414471   1833711623             113   
567  train_landmarks/5414471.parquet  5414471   1833797349             236   
569  train_landmarks/5414471.parquet  5414471   1833833168             151   
573  train_landmarks/5414471.parquet  5414471   1834088860             107   
576  train_landmarks/5414471.parquet  5414471   1834240610              18   
577  train_landmarks/5414471.parquet  5414471   1834277326             128   
583  train_landmarks/5414471.parquet  5414471   1834476149              59   
590  train_landmarks/5414471.parquet  5414471   1834686470             231   
591  train_landmarks/5414471.parquet  5414471   1834689645              95   
595  train_landmarks/5414471.parquet  5414471   1834776243               6   
596  train_landmarks/5414471.parquet  5414471   1834801771             128   
604  train_landmarks/5414471.parquet  5414471   1835174556               4   
611  train_landmarks/5414471.parquet  5414471   1835378367              88   
612  train_landmarks/5414471.parquet  5414471   1835389716              73   
615  train_landmarks/5414471.parquet  5414471   1835461523             157   
616  train_landmarks/5414471.parquet  5414471   1835553499             254   
619  train_landmarks/5414471.parquet  5414471   1835677830             227   
621  train_landmarks/5414471.parquet  5414471   1835753649             135   
626  train_landmarks/5414471.parquet  5414471   1836071941              59   
628  train_landmarks/5414471.parquet  5414471   1836177994             112   
629  train_landmarks/5414471.parquet  5414471   1836192092             151   
630  train_landmarks/5414471.parquet  5414471   1836202282              70   
632  train_landmarks/5414471.parquet  5414471   1836280427              15   
633  train_landmarks/5414471.parquet  5414471   1836287775             192   
636  train_landmarks/5414471.parquet  5414471   1836349635             186   
638  train_landmarks/5414471.parquet  5414471   1836386984              18   
639  train_landmarks/5414471.parquet  5414471   1836391909              76   
641  train_landmarks/5414471.parquet  5414471   1836478877             202   
645  train_landmarks/5414471.parquet  5414471   1836562724              89   
653  train_landmarks/5414471.parquet  5414471   1836797597              73   
656  train_landmarks/5414471.parquet  5414471   1836932619             187   
659  train_landmarks/5414471.parquet  5414471   1837044497               1   
665  train_landmarks/5414471.parquet  5414471   1837240959             143   
668  train_landmarks/5414471.parquet  5414471   1837326771             136   

                             phrase  
529            

After looking at the first 300 values, almost every case where a number is followed by a space could be considered as an adress, specially for our machine learning. 

In [57]:
rows_adresse = df3[df3['phrase'].str.match(r'^\d+\s')]

# To display the resulting rows:
rows_adresse

path     file_id  sequence_id  \
0         train_landmarks/5414471.parquet     5414471   1816796431   
2         train_landmarks/5414471.parquet     5414471   1816909464   
3         train_landmarks/5414471.parquet     5414471   1816967051   
4         train_landmarks/5414471.parquet     5414471   1817123330   
7         train_landmarks/5414471.parquet     5414471   1817171518   
...                                   ...         ...          ...   
67192  train_landmarks/2118949241.parquet  2118949241    387639942   
67196  train_landmarks/2118949241.parquet  2118949241    387891436   
67200  train_landmarks/2118949241.parquet  2118949241    387998966   
67206  train_landmarks/2118949241.parquet  2118949241    388235284   
67207  train_landmarks/2118949241.parquet  2118949241    388332538   

       participant_id                     phrase  
0                 217               3 creekhouse  
2                   1        1383 william lanier  
3                  63          988 franklin lane  
4                  89  6920 northeast 661st road  
7                 202       239613 stolze street  
...               ...                        ...  
67192              71    7133 cll rafael nazario  
67196             160      650895 old evensville  
67200              89  681 east 62nd place south  
67206              36          90 kerwood circle  
67207             176                 802 co 66b  

[21344 rows x 5 columns]

In [58]:
rows_adresse.head(50)

path  file_id  sequence_id  participant_id  \
0    train_landmarks/5414471.parquet  5414471   1816796431             217   
2    train_landmarks/5414471.parquet  5414471   1816909464               1   
3    train_landmarks/5414471.parquet  5414471   1816967051              63   
4    train_landmarks/5414471.parquet  5414471   1817123330              89   
7    train_landmarks/5414471.parquet  5414471   1817171518             202   
9    train_landmarks/5414471.parquet  5414471   1817216847              93   
20   train_landmarks/5414471.parquet  5414471   1817581380              15   
21   train_landmarks/5414471.parquet  5414471   1817606833             254   
22   train_landmarks/5414471.parquet  5414471   1817631401             169   
26   train_landmarks/5414471.parquet  5414471   1817711982              70   
28   train_landmarks/5414471.parquet  5414471   1817740422             141   
29   train_landmarks/5414471.parquet  5414471   1817770333             109   
32   train_landmarks/5414471.parquet  5414471   1817815567             230   
33   train_landmarks/5414471.parquet  5414471   1817882602             254   
36   train_landmarks/5414471.parquet  5414471   1817947249             109   
38   train_landmarks/5414471.parquet  5414471   1817969676             236   
39   train_landmarks/5414471.parquet  5414471   1818029273             217   
41   train_landmarks/5414471.parquet  5414471   1818053931             230   
44   train_landmarks/5414471.parquet  5414471   1818160881              89   
45   train_landmarks/5414471.parquet  5414471   1818211300             202   
46   train_landmarks/5414471.parquet  5414471   1818216000              56   
50   train_landmarks/5414471.parquet  5414471   1818249176             231   
58   train_landmarks/5414471.parquet  5414471   1818700520             236   
59   train_landmarks/5414471.parquet  5414471   1818712825             225   
65   train_landmarks/5414471.parquet  5414471   1818863391               0   
66   train_landmarks/5414471.parquet  5414471   1818864430               1   
70   train_landmarks/5414471.parquet  5414471   1818895313               9   
72   train_landmarks/5414471.parquet  5414471   1818992993             216   
77   train_landmarks/5414471.parquet  5414471   1819185262             151   
78   train_landmarks/5414471.parquet  5414471   1819251883             157   
82   train_landmarks/5414471.parquet  5414471   1819405994               2   
84   train_landmarks/5414471.parquet  5414471   1819442021              92   
85   train_landmarks/5414471.parquet  5414471   1819442901             225   
87   train_landmarks/5414471.parquet  5414471   1819496260             115   
91   train_landmarks/5414471.parquet  5414471   1819576828             187   
96   train_landmarks/5414471.parquet  5414471   1819645763              81   
100  train_landmarks/5414471.parquet  5414471   1819772594             239   
105  train_landmarks/5414471.parquet  5414471   1819973720             192   
108  train_landmarks/5414471.parquet  5414471   1820052645               2   
110  train_landmarks/5414471.parquet  5414471   1820068457             154   
111  train_landmarks/5414471.parquet  5414471   1820069004             188   
112  train_landmarks/5414471.parquet  5414471   1820149560               1   
113  train_landmarks/5414471.parquet  5414471   1820158385             171   
114  train_landmarks/5414471.parquet  5414471   1820256582             157   
117  train_landmarks/5414471.parquet  5414471   1820389021              80   
123  train_landmarks/5414471.parquet  5414471   1820502252             242   
124  train_landmarks/5414471.parquet  5414471   1820509466              20   
126  train_landmarks/5414471.parquet  5414471   1820536866              93   
127  train_landmarks/5414471.parquet  5414471   1820542985             145   
129  train_landmarks/5414471.parquet  5414471   1820582982               9   

                             phrase  
0              

In [67]:
# To remove rows where 'phrase' starts with a number or numbers followed by a space:
df4 = df3[~df3['phrase'].str.match(r'^\d+\s')]

# To display the resulting DataFrame after filtering:
df4.head(50)

path  file_id  sequence_id  participant_id  \
1    train_landmarks/5414471.parquet  5414471   1816825349             107   
11   train_landmarks/5414471.parquet  5414471   1817221621             153   
12   train_landmarks/5414471.parquet  5414471   1817229827             236   
15   train_landmarks/5414471.parquet  5414471   1817348598              20   
17   train_landmarks/5414471.parquet  5414471   1817370426              21   
24   train_landmarks/5414471.parquet  5414471   1817657937               2   
35   train_landmarks/5414471.parquet  5414471   1817940885               0   
55   train_landmarks/5414471.parquet  5414471   1818613605             136   
56   train_landmarks/5414471.parquet  5414471   1818674500             196   
60   train_landmarks/5414471.parquet  5414471   1818726340             186   
62   train_landmarks/5414471.parquet  5414471   1818787239             161   
63   train_landmarks/5414471.parquet  5414471   1818802303             230   
67   train_landmarks/5414471.parquet  5414471   1818872649             136   
69   train_landmarks/5414471.parquet  5414471   1818887623             168   
83   train_landmarks/5414471.parquet  5414471   1819437603             121   
86   train_landmarks/5414471.parquet  5414471   1819492087              59   
89   train_landmarks/5414471.parquet  5414471   1819539423             141   
90   train_landmarks/5414471.parquet  5414471   1819561171              36   
95   train_landmarks/5414471.parquet  5414471   1819642365              74   
119  train_landmarks/5414471.parquet  5414471   1820405839             154   
120  train_landmarks/5414471.parquet  5414471   1820406063             223   
122  train_landmarks/5414471.parquet  5414471   1820470531             169   
125  train_landmarks/5414471.parquet  5414471   1820510214              92   
128  train_landmarks/5414471.parquet  5414471   1820564392             113   
133  train_landmarks/5414471.parquet  5414471   1820762462             196   
140  train_landmarks/5414471.parquet  5414471   1821071725             161   
154  train_landmarks/5414471.parquet  5414471   1821617529             202   
156  train_landmarks/5414471.parquet  5414471   1821859121             145   
160  train_landmarks/5414471.parquet  5414471   1822015644             107   
164  train_landmarks/5414471.parquet  5414471   1822165432             236   
173  train_landmarks/5414471.parquet  5414471   1822600075              95   
174  train_landmarks/5414471.parquet  5414471   1822605218             112   
179  train_landmarks/5414471.parquet  5414471   1822774644             112   
180  train_landmarks/5414471.parquet  5414471   1822969244              21   
181  train_landmarks/5414471.parquet  5414471   1822991532             192   
183  train_landmarks/5414471.parquet  5414471   1823093477              68   
191  train_landmarks/5414471.parquet  5414471   1823397819              24   
203  train_landmarks/5414471.parquet  5414471   1823749864              24   
209  train_landmarks/5414471.parquet  5414471   1823933990               4   
213  train_landmarks/5414471.parquet  5414471   1824019526             181   
215  train_landmarks/5414471.parquet  5414471   1824056951             188   
232  train_landmarks/5414471.parquet  5414471   1824476074              21   
236  train_landmarks/5414471.parquet  5414471   1824640811             157   
242  train_landmarks/5414471.parquet  5414471   1824817891             151   
250  train_landmarks/5414471.parquet  5414471   1825064247             107   
251  train_landmarks/5414471.parquet  5414471   1825082449             112   
262  train_landmarks/5414471.parquet  5414471   1825332072             219   
266  train_landmarks/5414471.parquet  5414471   1825419903              56   
268  train_landmarks/5414471.parquet  5414471   1825493767              92   
269  train_landmarks/5414471.parquet  5414471   1825500764             192   

                             phrase  
1              

### Some outliers findings : 

In [64]:
rows_with_at_symbol = df3[df3['phrase'].str.contains('@')]

# To display the resulting rows:
rows_with_at_symbol

path     file_id  sequence_id  \
2975    train_landmarks/128822441.parquet   128822441    227635292   
3460    train_landmarks/149822653.parquet   149822653   1542053334   
10043   train_landmarks/388576474.parquet   388576474   1055702633   
11517   train_landmarks/425182931.parquet   425182931   1131878382   
12982   train_landmarks/433948159.parquet   433948159    699141994   
15984   train_landmarks/522550314.parquet   522550314    787227070   
23285   train_landmarks/649779897.parquet   649779897   1272562464   
24553   train_landmarks/654436541.parquet   654436541   1664169961   
25628   train_landmarks/683666742.parquet   683666742   1697177280   
28135   train_landmarks/882979387.parquet   882979387    193803180   
29492   train_landmarks/939623093.parquet   939623093   1758946625   
32096  train_landmarks/1021040628.parquet  1021040628   1747065771   
33437  train_landmarks/1099408314.parquet  1099408314   1215702881   
34725  train_landmarks/1133664520.parquet  1133664520    557085009   
39066  train_landmarks/1341528257.parquet  1341528257   1425288262   
39101  train_landmarks/1341528257.parquet  1341528257   1426390791   
44062  train_landmarks/1448136004.parquet  1448136004   2035998238   
44301  train_landmarks/1497621680.parquet  1497621680   1022117355   
45122  train_landmarks/1497621680.parquet  1497621680   1049946092   
45663  train_landmarks/1552432300.parquet  1552432300    809432369   
45885  train_landmarks/1552432300.parquet  1552432300    816201203   
47863  train_landmarks/1562234637.parquet  1562234637    946334679   
48364  train_landmarks/1643479812.parquet  1643479812    392119245   
49778  train_landmarks/1647220008.parquet  1647220008   1481090463   
50643  train_landmarks/1662742697.parquet  1662742697   1351039125   
54110  train_landmarks/1785039512.parquet  1785039512   1650265110   
56782  train_landmarks/1905462118.parquet  1905462118   1164241701   
56925  train_landmarks/1905462118.parquet  1905462118   1169120447   
59490  train_landmarks/1967755728.parquet  1967755728    997820966   
62383  train_landmarks/2026717426.parquet  2026717426   1184110736   
62883  train_landmarks/2026717426.parquet  2026717426   1199040800   
64556  train_landmarks/2072296290.parquet  2072296290     77276729   
65731  train_landmarks/2072876091.parquet  2072876091   1960269268   
66730  train_landmarks/2118949241.parquet  2118949241    372167985   

       participant_id                          phrase  
2975               81        @dr.karanr/plainfield-wi  
3460              107      /abakaliki/magamrol/@y1991  
10043             112                    @diegopirett  
11517              56           pliers/@sarahmagusara  
12982             225            /@anita_betun26/8546  
15984             105       tierarzt/@karlaitcamberos  
23285              21            /899/@marikanortonx/  
24553             236            /@anita_betun26/8546  
25628               1                    @lil_sammyt/  
28135              76            /899/@marikanortonx/  
29492             143                    @diegopirett  
32096              89              naoslibros/@payton  
33437             178  takasaki.keizai.biz/@gabinofit  
34725             161                      @antaios:b  
39066             233   imperia/@wongbraling/ganglion  
39101             125         /@elesslove/minneapolis  
44062             105          @xingailuyinqun/yanshi  
44301             254                      @antaios:b  
45122             160   imperia/@wongbraling/ganglion  
45663              68  takasaki.keizai.biz/@gabinofit  
45885             115                     @goodnight/  
47863               1            /@anita_betun26/8546  
48364             178    mechas-frontales/@artwithflo  
49778             115   imperia/@wongbraling/ganglion  
50643               9                   @sgtaylor/816  
54110             233                     @goodnight/  
56782              81        @dr.karanr/plainfield-wi  
56925             

They don't seem to be adresses but rather identifiant from a social network such as Instragram.

## Next Step : Now let's see the frequency of each character. Will be useful as we are looking at fingerspelling in ASL we have to know what we will feed our machine learning to learn from.

In [16]:
df = pd.read_csv('d:/Bureau/ASLCapstone/train.csv/train.csv')
df

path     file_id  sequence_id  \
0         train_landmarks/5414471.parquet     5414471   1816796431   
1         train_landmarks/5414471.parquet     5414471   1816825349   
2         train_landmarks/5414471.parquet     5414471   1816909464   
3         train_landmarks/5414471.parquet     5414471   1816967051   
4         train_landmarks/5414471.parquet     5414471   1817123330   
...                                   ...         ...          ...   
67203  train_landmarks/2118949241.parquet  2118949241    388192924   
67204  train_landmarks/2118949241.parquet  2118949241    388225542   
67205  train_landmarks/2118949241.parquet  2118949241    388232076   
67206  train_landmarks/2118949241.parquet  2118949241    388235284   
67207  train_landmarks/2118949241.parquet  2118949241    388332538   

       participant_id                          phrase  
0                 217                    3 creekhouse  
1                 107                 scales/kuhaylah  
2                   1             1383 william lanier  
3                  63               988 franklin lane  
4                  89       6920 northeast 661st road  
...               ...                             ...  
67203              88                    431-366-2913  
67204             154                    994-392-3850  
67205              95  https://www.tianjiagenomes.com  
67206              36               90 kerwood circle  
67207             176                      802 co 66b  

[67208 rows x 5 columns]

In [20]:
# Dictionary to store counts of each digit
digit_counts = {}

# Loop through each digit from 0 to 9
for digit in range(10):
    # Convert the digit to a string and use 'str.count()' to count occurrences in the 'phrase' column
    count_of_digit = df['phrase'].str.count(str(digit)).sum()
    # Store the count in the dictionary with the digit as the key
    digit_counts[str(digit)] = count_of_digit

# Print the counts of each digit
for digit, count in digit_counts.items():
    print("Number of occurrences of '{}' in the 'phrase' column: {}".format(digit, count))

Number of occurrences of '0' in the 'phrase' column: 26744
Number of occurrences of '1' in the 'phrase' column: 31292
Number of occurrences of '2' in the 'phrase' column: 31599
Number of occurrences of '3' in the 'phrase' column: 29789
Number of occurrences of '4' in the 'phrase' column: 29810
Number of occurrences of '5' in the 'phrase' column: 29565
Number of occurrences of '6' in the 'phrase' column: 29701
Number of occurrences of '7' in the 'phrase' column: 28119
Number of occurrences of '8' in the 'phrase' column: 28202
Number of occurrences of '9' in the 'phrase' column: 29440


In [22]:
# Dictionary to store counts of each letter
letter_counts = {}

# Loop through each letter of the alphabet (both uppercase and lowercase)
for letter in 'fghijklmnopqrstuvwxyz':
    # Use 'str.count()' to count occurrences of the letter in the 'phrase' column
    count_of_letter = df['phrase'].str.count(letter).sum()
    # Store the count in the dictionary with the letter as the key
    letter_counts[letter] = count_of_letter

# Print the counts of each letter
for letter, count in letter_counts.items():
    print("Number of occurrences of '{}' in the 'phrase' column: {}".format(letter, count))

Number of occurrences of 'f' in the 'phrase' column: 8639
Number of occurrences of 'g' in the 'phrase' column: 15257
Number of occurrences of 'h' in the 'phrase' column: 24243
Number of occurrences of 'i' in the 'phrase' column: 43840
Number of occurrences of 'j' in the 'phrase' column: 4403
Number of occurrences of 'k' in the 'phrase' column: 12355
Number of occurrences of 'l' in the 'phrase' column: 38452
Number of occurrences of 'm' in the 'phrase' column: 26391
Number of occurrences of 'n' in the 'phrase' column: 45663
Number of occurrences of 'o' in the 'phrase' column: 57784
Number of occurrences of 'p' in the 'phrase' column: 17326
Number of occurrences of 'q' in the 'phrase' column: 1114
Number of occurrences of 'r' in the 'phrase' column: 55533
Number of occurrences of 's' in the 'phrase' column: 43224
Number of occurrences of 't' in the 'phrase' column: 47132
Number of occurrences of 'u' in the 'phrase' column: 20907
Number of occurrences of 'v' in the 'phrase' column: 10370


In [25]:
char_counts = {}

# List of characters to count occurrences
characters = ["~", "", "!", "#", "$", "%", "&", "'", "(", ")", "*", "+", ",", "-", ".", "/", ":", ";", "=", "?", "@", "[", "_"]

# Loop through each character
for char in characters:
    # Escape the special characters in the regular expression pattern
    escaped_char = re.escape(char)
    # Use 'str.count()' to count occurrences of the character in the 'phrase' column
    count_of_char = df['phrase'].str.count(escaped_char).sum()
    # Store the count in the dictionary with the character as the key
    char_counts[char] = count_of_char

# Print the counts of each character
for char, count in char_counts.items():
    print("Number of occurrences of '{}' in the 'phrase' column: {}".format(char, count))

Number of occurrences of '~' in the 'phrase' column: 21
Number of occurrences of '' in the 'phrase' column: 1263822
Number of occurrences of '!' in the 'phrase' column: 3
Number of occurrences of '#' in the 'phrase' column: 3
Number of occurrences of '$' in the 'phrase' column: 5
Number of occurrences of '%' in the 'phrase' column: 128
Number of occurrences of '&' in the 'phrase' column: 24
Number of occurrences of ''' in the 'phrase' column: 3
Number of occurrences of '(' in the 'phrase' column: 16
Number of occurrences of ')' in the 'phrase' column: 16
Number of occurrences of '*' in the 'phrase' column: 3
Number of occurrences of '+' in the 'phrase' column: 8307
Number of occurrences of ',' in the 'phrase' column: 40
Number of occurrences of '-' in the 'phrase' column: 54242
Number of occurrences of '.' in the 'phrase' column: 24652
Number of occurrences of '/' in the 'phrase' column: 19477
Number of occurrences of ':' in the 'phrase' column: 1939
Number of occurrences of ';' in the

<!-- This is at the bottom of the file -->
<a id="bottom"></a>

### Let's build up the ``dictionary of frequency`` : 

``Numbers : ``

"0": 26 744       

"1": 31 292 

"2": 31 599 

"3": 29 789 

"4": 29 810 

"5": 29 565 

"6": 29 701 

"7": 28 119 

"8": 28 202

"9": 29 440 

``Alphabet : ``

'a': 69521

'b': 13655

'c': 33020

'd': 25751

'e': 71986

'f': 8639

'g': 15257

'h': 24243

'i': 43840

'j': 4403

'k': 12355

'l': 38452

'm': 26391

'n': 45663

'o': 57784

'p': 17326

'q': 1114

'r': 55533

's': 43224

't': 47132

'u': 20907

'v': 10370

'w': 28532

'x': 2315

'y': 11409

'z' : 3746

``Special character : ``

'': 1263822

'-': 54242

'.': 24652

'/': 19477

'+': 8307

':': 1939

'_': 1683

'=': 304

'?': 302

'%': 128

'@': 45

',': 40

'&': 24

'~': 21

'(': 16

')': 16

'$': 5

'!': 3

'#': 3

'*': 3

'\'': 3

'[': 2

';': 1